src/test/java/me/karakelley/http/HttpServerTest.java

```diff
@@ -277,7 +277,37 @@ void testDisplaysFilesAtRoot() {
         e.printStackTrace();
       }
       assertTrue(response.contains("Content-Type: text/html"));
-      assertTrue(response.contains("<p>"+fileOne.getFileName()+"</p>"));
+      assertTrue(response.contains("<p><a href=\"/test1.txt\">test1.txt</a></p>"));
+    });
+  }
+
+  @Test
+  void testReturnsFileContentsRequested() {
+    TempFilesHelper.withTempDirectory(directory -> {
+      Path fileOne = TempFilesHelper.createTempFile(directory, "/test1");
+      writeToFile("Hello World", fileOne);
+      ServerConfiguration config = new ServerConfiguration();
+      config.setController(new Application(PublicDirectory.create(directory.toString(), new FileFinderCache(new RealFileFinder()))));
+      config.setPort("0");
+      HttpServer httpServer = new HttpServer(config);
+      ClientHelper client = new ClientHelper();
+
+      ExecutorService executorService = Executors.newSingleThreadExecutor();
+      executorService.submit(httpServer::start);
+      try {
+        client.connectWithTry("127.0.0.1", httpServer);
+      } catch (InterruptedException | IOException e) {
+        e.printStackTrace();
+      }
+
+      ArrayList<String> response = null;
+      try {
+        response = client.sendMessage("GET /test1.txt HTTP/1.1\r\n");
+      } catch (IOException e) {
```

Comment:
Does this ever get thrown and caught when you run your tests?

Reply:
Not that I've noticed - I kept getting compiler errors without the `try/catch` block within the lambda and adding it to the method signature wouldn't work either.

Reply:
Fair enough, this one might be unavoidable. Still curious about some of the other `try/catch` blocks I commented on.